docs/media.md

```diff
@@ -0,0 +1,161 @@
+example at FUN, we use 2 separate OpenStack projects: `fun_development` and
+`fun_production`.
+
+Place the `openrc.sh` file for the admin user under this directory.
+
+> Note that you will be able to create several media files buckets per
+> OpenStack project in case you want to host several environments in the same
+> project. For example at FUN, the following environments are all in the
+> `fun_development` project: `staging`, `preprod`, `PR branches`,...
+
+Finally, edit the `openrc.sh` file and add the following variable with the
+name of the second user you created for Django (by name we mean Keystone's
+username).
+
+```
+export TF_VAR_user_name="joanie-user"
+```
+
+Your complete `env.d/terraform/production/openrc.sh` file should look
+something like:
+
+```
+#!/bin/bash
+
+# To use an Openstack cloud you need to authenticate against keystone, which
+# returns a **Token** and **Service Catalog**. The catalog contains the
+# endpoint for all services the user/tenant has access to - including nova,
+# glance, keystone, swift.
+#
+export OS_AUTH_URL=https://auth.cloud.ovh.net/v3/
+export OS_IDENTITY_API_VERSION=3
+
+export OS_USER_DOMAIN_NAME=${OS_USER_DOMAIN_NAME:-"Default"}
+export OS_PROJECT_DOMAIN_NAME=${OS_PROJECT_DOMAIN_NAME:-"Default"}
+
+
+# With the addition of Keystone we have standardized on the term **tenant**
+# as the entity that owns the resources.
+export OS_TENANT_ID=f5h78jko9aex435gt3d4rfg76n1a3ze4
+export OS_TENANT_NAME=2643565598702422
+
+# In addition to the owning entity (tenant), openstack stores the entity
+# performing the action as the **user**.
+export OS_USERNAME="my-openstack-admin-user"
+
+# With Keystone you pass the keystone password.
+echo "Please enter your OpenStack Password: "
+read -sr OS_PASSWORD_INPUT
+export OS_PASSWORD=$OS_PASSWORD_INPUT
+
+# If your configuration has multiple regions, we set that information here.
+# OS_REGION_NAME is optional and only valid in certain environments.
+export OS_REGION_NAME="GRA"
+# Don't leave a blank variable, unset it if it was empty
+if [ -z "$OS_REGION_NAME" ]; then unset OS_REGION_NAME; fi
+
+# Configure the name of the Django user that will connect to the media bucket
+export TF_VAR_user_name="joanie-user"
+```
+
+## Setup a shared Terraform state
+
+> ✋ If the project already exists with a shared state, you should skip this
+> section and start fetching the state locally to apply changes.
+
+If the project doesn't exist at all, you will need to create a Swift bucket
+to store your Terraform state file by typing the following commands in your
+terminal:
+
+```
+$ bin/state init
+$ bin/state apply
+```
+
+And voilà! Your shared state is now available to anyone contributing to the
+project. It was placed in a bucket named `joanie-terraform` and a backup
+bucket named `joanie-terraform-archive` was also created to historize all
+changes.
+
+> Note that there is no lock mechanism, so you should make sure changes are
+> not applied by 2 persons simultaneously.
```

Comment:
[Swift](https://www.terraform.io/language/settings/backends/swift) backend supports [state locking](https://www.terraform.io/language/state/locking). May be we should warn user to take care to check if the backend he used support state locking.

Reply:
but in our implementation we make the choice of Swift as a backend... which does not support locking...
Maybe we should not use S3/Swift for states but my assumption is that there is no other solution for shared backend that is as universal and easy?

Reply:
Oh ! I believed I mixed concepts of media storage backend and terraform backend. So ok the storage backend does not support lock mechanism but as we manage access to swift bucket through terraform there is a lock mechanism right?

Reply:
Oh you're right :open_mouth: I just tested it and it works.
I had just extrapolated from the S3 backend which requires dynamoDB to support state locking...
So I will now remove this sentence from documentation :clap: